display_list/display_list_utils.cc

```diff
@@ -423,12 +424,24 @@ void DisplayListBoundsCalculator::setPathEffect(const DlPathEffect* effect) {
 void DisplayListBoundsCalculator::setMaskFilter(const DlMaskFilter* filter) {
   mask_filter_ = filter ? filter->shared() : nullptr;
 }
-void DisplayListBoundsCalculator::save() {
+
+void DisplayListBoundsCalculator::doSave() {
   SkMatrixDispatchHelper::save();
   ClipBoundsDispatchHelper::save();
+  current_layer()->deferred_save_count_ -= 1;
   layer_infos_.emplace_back(std::make_unique<LayerData>(nullptr));
   accumulator_.save();
 }
+
+void DisplayListBoundsCalculator::checkForDeferredSave() {
+  if (current_layer()->deferred_save_count_ > 0) {
+    doSave();
+  }
+}
+
+void DisplayListBoundsCalculator::save() {
```

Comment:
deferred save elimination was already done during construction. Why is it being done here as well? The stream should already be reduced so all of this code in the calculator should be completely redundant. Does it actually encounter any missed deferred save/restores?

Reply:
This still seems unnecessary and yet it is still here. Why are you checking the save/restores in the Calculator? That operation was already done in the Builder.